flow/Net2.actor.cpp

```diff
@@ -604,9 +652,22 @@ class SSLConnection : public IConnection, ReferenceCounted<SSLConnection> {
 			int64_t permitNumber = wait(g_network->networkInfo.handshakeLock->take());
 			state BoundedFlowLock::Releaser releaser(g_network->networkInfo.handshakeLock, permitNumber);
 
-			BindPromise p("N2_AcceptHandshakeError", UID());
-			auto onHandshook = p.getFuture();
-			self->getSSLSocket().async_handshake( boost::asio::ssl::stream_base::server, std::move(p) );
+			Future<Void> onHandshook;
+
+			// If the background handshakers are not all busy, use one
+			if(N2::g_net2->sslPoolHandshakesInProgress < FLOW_KNOBS->TLS_HANDSHAKE_THREADS) {
```

Comment:
This feature can be disabled by setting the knob to zero, so it should be fairly low risk.

Reply:
That's the plan, and a good way to test the fallback behavior!